mk/config.mk

```diff
@@ -984,13 +984,13 @@ endif
 CFG_WDT ?= n
 
 # Enable watchdog SMC handling compatible with arm-smc-wdt Linux driver
-# When enabled, CFG_WDT_SM_HANDLER_ID must be defined with a SMC ID
+# When CFG_WDT_SM_HANDLER=y, SMC function ID 0xb2000014 implements
+# arm-smc-wdt service. Platform can also define an additional platform
+# specific SMC function ID to access arm-smc-wdt service thanks to optional
+# config switch CFG_WDT_SM_HANDLER_ID.
 CFG_WDT_SM_HANDLER ?= n
 
 $(eval $(call cfg-enable-all-depends,CFG_WDT_SM_HANDLER,CFG_WDT))
-ifeq (y-,$(CFG_WDT_SM_HANDLER)-$(CFG_WDT_SM_HANDLER_ID))
-$(error CFG_WDT_SM_HANDLER_ID must be defined when enabling CFG_WDT_SM_HANDLER)
```

Comment:
I don't see why don't always require `CFG_WDT_SM_HANDLER_ID` to define the id.

Reply:
SMC funcID  defined with `CFG_WDT_SM_HANDLER_ID` is trapped by OP-TEE Monitor (_core/arch/arm/sm/sm.c_). The proposed change allows to route Watchdog SMC functions to OP-TEE even when using TF-A BL31 Monitor.

Reply:
@etienne-lms is too fast. Confirm the reply too.

Reply:
That doesn't answer the question. Why can't we always use `CFG_WDT_SM_HANDLER_ID` to define the watchdog SMC ID? We could define a sensible default value, like `0x82003D06`. If it's not good enough for some platform, just override it.

Reply:
Defining this one will make it a standard that is not better that giving a OP-TEE SMC TOS ID instead. Only working again for armv7 not supported on armv8 64bit.

Reply:
Use whatever default value you find useful, but it should be defined as the SMC ID used in the current configuration, and platforms are supposed to be able to override it. No need for a second way to define the SMC ID.